paperwork-backend/paperwork_backend/docsearch.py

```diff
@@ -264,16 +264,20 @@ def __init__(self, rootdir, indexdir=None, language=None,
 
         self.fs = fs.GioFileSystem()
         self.rootdir = self.fs.safe(rootdir)
-
         localdir = os.path.expanduser("~/.local")
-        base_data_dir = os.getenv(
-            "XDG_DATA_HOME",
-            os.path.join(localdir, "share")
-        )
-        if indexdir is None:
-            indexdir = os.path.join(base_data_dir, "paperwork")
-
-        indexdir = os.path.join(indexdir, "index")
+
+        if index_in_workdir:
+            base_data_dir = self.fs.unsafe(rootdir)
+            localdir = base_data_dir
+            indexdir = os.path.join(base_data_dir, "index")
+
+        elif indexdir is None:
+            base_data_dir = os.getenv(
```

Comment:
if `not index_in_workdir` and `indexdir is not None` then `base_data_dir` won't be set, but still be passed to `self.index.open()`.

Reply:
I corrected this by moving the base_data_dir = self.fs.unsafe(rootdir) to before the "if index_in_workdir" statement.